server/aesdsocket.c

```diff
@@ -79,6 +134,113 @@ void write_to_file(int fd, void *buf, size_t buf_size)
+}
+
+
+void* thread_foo(void *thread_param)
+{
+
+    struct thread_data_s* thread_func_args = (struct thread_data_s *) thread_param;
+    reading = true;
+    // inet_ntop(their_addr.ss_family,
+    //     get_in_addr((struct sockaddr *)&their_addr),
+    //     s, sizeof s);
+    // printf("server: got connection from %s\n", s);
+
+    //int local_fd = open("/var/tmp/aesdsocketdata", O_CREAT | O_RDWR | O_APPEND, S_IRWXU |  S_IRWXG | S_IRWXO );
+    // int local_fd = open("/var/tmp/aesdsocketdata", O_CREAT | O_RDWR | O_APPEND, 777);
+    // if(local_fd < 0)
+    //     perror("open()");
+    printf("file opened\n");
+    int max_buf_size = BUF_SIZE;
+    char *buf = (char *)malloc(max_buf_size * sizeof(char));
+    int return_idx = 0;
+    int num_of_relloc = 0;
+    bool ret_found = false;
+
+    while ( !ret_found ) //return not found
+    {
+        if(recv(thread_func_args->sfd, buf + (num_of_relloc * BUF_SIZE), BUF_SIZE, 0) == - 1){
+            //log errors
+            perror("recv()");
+        }
+
+        return_idx = search_for_return(buf + (num_of_relloc * BUF_SIZE), BUF_SIZE);
+        if(return_idx == -1)
+        {
+            max_buf_size += BUF_SIZE;
+            buf = (char *)realloc(buf, max_buf_size * sizeof(char));
+            num_of_relloc++;
+        }
+        else
+        {
+            printf("writing to file\n");     
+            write_to_file(thread_func_args->local_fd, buf, max_buf_size - (BUF_SIZE - return_idx));
+            ret_found = true;
+            break;
+        }
+    }
+
+    int num_of_bytes = BUF_SIZE;
+
+    pthread_mutex_lock(&mutex_lock);
+    if (lseek(thread_func_args->local_fd, 0, SEEK_SET) < 0)
+        perror("lseek()");
+    pthread_mutex_unlock(&mutex_lock);
+
+    char read_buf[BUF_SIZE];
+
+    printf("Attemping to read and send BUF_SIZE\n");
+    while(num_of_bytes == BUF_SIZE)
+    {
+        pthread_mutex_lock(&mutex_lock);
+        num_of_bytes = read(thread_func_args->local_fd, read_buf, BUF_SIZE);
+        if(num_of_bytes == -1)
+            perror("read()");
+        pthread_mutex_unlock(&mutex_lock);
+        printf("num of bytes: %d\n", num_of_bytes);
+        send(thread_func_args->sfd , read_buf, num_of_bytes, 0);
+    }
+    
+    free(buf);
+    reading = false;
+    thread_func_args->thread_complete_success =  true;
+
+    printf("thread completed\n");
+
+
+    if(close(thread_func_args->sfd) < 0)
+        perror("close()");
+
+  
+    //return thread_param;
+    return 0;
+
+}
+
+
+void timer_handler()
+{
+    printf ("10 Seconds passed\n");
+
+    time_elasped++;
+
+
+    // size_t strftime(char *restrict s, size_t max,
+    //                    const char *restrict format,
+    //                    const struct tm *restrict tm);
+    time_t rawtime;
+    struct tm *info;
+    //char buffer[80];
+
+    time( &rawtime );
+    char time_val[40];
+    info = localtime( &rawtime );
+    strftime(time_val,40,"%Y/%m/%d - %H:%M:%S", info);
```

Comment:
@PeterBraganza don't hard-code 40 here and at line 239, add a define or const variable for this.